### PR TITLE
form processing error logic cleanup

### PR DESCRIPTION
**Pull request by snopoke**

### CODE REVIEW

File: corehq/form_processor/submission_post.py [hunk 1]
```diff
@@ -307,22 +307,13 @@ def _get_open_rosa_response(self, instance, errors):
...
-                status=201,
-            ).response()
-        else:
-            response = OpenRosaResponse(
-                # would have done ✓ but our test Nokias' fonts don't have that character
-                message=u'   √   ',
-                nature=ResponseNature.SUBMIT_SUCCESS,
-                status=201,
-            ).response()
-        return response
```
Comment (benrudolph) [inline corehq/form_processor/submission_post.py]: haha

**Review by benrudolph — APPROVED** [review]

**Comment by benrudolph** [comment]
One thought: we may want to tag known errors differently in datadog

**Comment by snopoke** [comment]
@proteusvacuum unknown errors don't get recorded because they raise an exception

**Comment by snopoke** [comment]
:white_check_mark:

**Comment by benrudolph** [comment]
was that comment for me? i was referring to known errors, but anyways seems fine to me

**Comment by benrudolph** [comment]
oh i see you're saying that all error types are known error types

**Pull request merged**